pkg/engine/runtime/kubernetes_runtime.go

```diff
@@ -194,21 +197,19 @@ func (k *KubernetesRuntime) Watch(ctx context.Context, request *WatchRequest) *W
 }
 
 // getKubernetesClient get kubernetes client
-func getKubernetesClient() (dynamic.Interface, *restmapper.DeferredDiscoveryRESTMapper, error) {
+func getKubernetesClient() (dynamic.Interface, meta.RESTMapper, error) {
 	// build config
 	cfg, err := clientcmd.BuildConfigFromFlags("", config.GetKubeConfig())
 	if err != nil {
 		return nil, nil, err
 	}
 
-	// Prepare a RESTMapper to find GVR
-	dc, err := discovery.NewDiscoveryClientForConfig(cfg)
+	// DynamicRESTMapper can discover resource types at runtime dynamically
+	mapper, err := apiutil.NewDynamicRESTMapper(cfg)
```

Comment:
What is the difference between NewDiscoveryClientForConfig() and NewDynamicRESTMapper(), since we only invoke this method once?

Reply:
see here: https://github.com/kubernetes-sigs/controller-runtime/pull/554/files#diff-2a0fbc494b5a946115f60dae4b372706b2a54a0eedd66f5d4e05b3584b164246